harrisonDir/harrisonHomework.txt

```diff
@@ -0,0 +1,271 @@
+1.) 
+
+var properFrac = function(n,d) {
+  if (n >= d && n%d != 0) {				/* If it's improper and n is not divisible by d*/
+    var wholeNum = ( Math.floor(n/d) );		/* Number of times n goes into d completely */
+    var remainder = n%d;
+    return ( wholeNum + " " + remainder + "/" + d ); 
+  } else if ( Math.floor(n/d) === 0) {
+      return (n/n) + "/" + (d/n);			/* If n goes into d evenly, simplify */
+  } else {								/* Any other cases should be proper fractions, so just put a slash between them */
+    	return (n + "/" + d);
+  }
+};
+
+2.)
+
+a) var i;
+
+Math.round(n);
+
+b) 
+
+var y;
+
+if (x) {
+	y = x;
+} else {
+	y = false;
+}
+
+c) for (var count = 15 - i; count < 15; count++) {
+	console.log(i);
+	i--;
+}
+
+d) 
+
+var x;
+if (a && b) {
+	x = 0;
+} else if (!a && !b) {
+	x = 2;
+} else {
+	x = 1;
+}
+
+3.) 
+a)
+
+var letterTriangle  = function(n) {
+	
+  var alphabet = ['A\n','B','C','D','E','F','G','H','I','J','K','L','M','N',
+                 'O','P','Q','R','S','T','U','V','W','X','Y','Z'];
+  
+  var fullString = 'A\n';  
+    
+  var currentLine = alphabet[0];
+  
+  for (var i = 1; i < n; i++) {	
+    var newLine = alphabet[i] + currentLine; //Always end on \n //
+    fullString = fullString + newLine;
+    currentLine = newLine;
+  }
+
+  console.log(fullString);
+};
+
+letterTriangle(26);
+
+b) var twelveDays  = function(n) {
+  
+  var gifts = [0,'and a partridge in a pear tree!\n','two turtle doves,\n','three French hens,\n','four calling birds,\n','five golden rings,\n','six geese a-laying,\n','seven swans a-swimming,\n','eight maids a-milking,\n','nine ladies dancing,\n','ten lords a-leaping,\n','eleven pipers piping,\n','twelve drummers drumming,\n']
+  
+  switch (n) {
+    case 1:
+      var fullString = 'On the 1st day of Christmas, my true love gave to me: \n';
+      break;
+    case 2:
+      var fullString = 'On the 2nd day of Christmas, my true love gave to me: \n';
+      break;
+    case 3:
+      var fullString = 'On the 3rd day of Christmas, my true love gave to me: \n';
+      break;
+    default: 
+      var fullString = 'On the ' + n + "th day of Christmas, my true love gave to me: \n";
+      break;
+  }
+  
+  var currentLine = gifts[n];
+  
+  for (var i = n; i > 0; i--) { 
+    var newLine = gifts[i]; //Always end on \n //
+    fullString = fullString + newLine;
+  }
+
+  console.log(fullString);
+};
+
+twelveDays(2);
+
```

Comment:
This is super close! When you console log different numbers from 1 to 12, your function "twelveDays" returns the correct "case" values. It doesn't, however, return anything from your "gifts" array, and it also doesn't log the entire song to the console. 

You would need to write the function in a way that both the gifts and days are returned. To make this easier, it would work well to place the "gifts" in an array, and also place the "days" in an array, like so: 

var ordinals = ['first','second','third','fourth','fifth','sixth',
            'seventh','eighth','ninth','tenth','eleventh','twelfth'];

var items = ["a partridge in a pear tree.\n",
            "two turtle doves, \nand ",
            "three french hens,\n",
            "four calling birds,\n",
            "five... gold... rings,\n",
            "six geese-a-laying,\n",
            "seven swans-a-swimming,\n",
            "eight maids-a-milking,\n",
            "nine ladies dancing,\n",
            "ten lords-a-leaping,\n",
            "eleven pipers piping,\n",
            "twelve drummers drumming,\n"]  

You would then want to write a function that returns each verse with the corresponding day, and you would want to utilize a couple of loops in order to do so. Below is Dan's solution to this problem: 

function generateDay(n) {
    var verse = "On the "
            +ordinals[n]
            +" day of Xmas, my true love gave to me\n";
    for (var i=n; i>=0; i--) {
        verse += items[i];
    }
    return verse;
}

function generateSong() {
    var song="";
    for (var day=0; day < 12; day++) {
        song += generateDay(day)+'\n'; //calls inner loop
    }
    return song;
}

console.log(generateSong());

You were definitely on the right track with this one, and demonstrated all of the necessary concepts needed to get the right answer. Nice work!!